app/eth2wrap/eth2wrap.go

```diff
@@ -305,3 +331,53 @@ func wrapError(ctx context.Context, err error, label string) error {
 
 	return errors.Wrap(err, "beacon api "+label, z.Str("label", label))
 }
+
+// newBestSelector returns a new bestSelector.
+func newBestSelector(n int, period time.Duration) *bestSelector {
+	return &bestSelector{
+		n:      n,
+		period: period,
+		counts: make([]int, n),
+	}
+}
+
+// bestSelector calculates the "best client index" per period.
+type bestSelector struct {
+	n      int
+	mu     sync.Mutex
+	start  time.Time
+	period time.Duration
+	counts []int
+}
+
+// Best returns the best index or 0 if no counts.
+func (s *bestSelector) Best() int {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	var resp, count int
```

Comment:
what is the purpose of the `count` variable? seems like it's never updated

Reply:
umm..i think i now understand what you're trying to do: `get the array index with the greatest value`